vcd/provider.go

```diff
@@ -246,6 +246,13 @@ func Provider() *schema.Provider {
 				Description: "The API token used instead of username/password for VCD API operations. (Requires VCD 10.3.1+)",
 			},
 
+			"service_account": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("VCD_SA_TOKEN_FILE", nil),
+				Description: "The Service Account API token file instead of username/password for VCD API operations. (Requires VCD 10.4.0+)",
```

Comment:
With `token`, `api_token`, `saml_adfs_rpt_id` and `service_account_token_file` (what did I miss?) parameters available now, it is becoming increasingly easier to mis-configure.

It seems, we do have runtime validation (is it enough?):
https://github.com/vmware/terraform-provider-vcd/pull/1040/files#diff-e96a5cf811e4de4a40ed325399541cc28aef6972ab94e260331cb58898cd6a8bR366-R369

But it looks like the `auth_type` doc section needs to be improved:
https://registry.terraform.io/providers/vmware/vcd/latest/docs#auth_type

Specifically:
- `saml_adfs` should state which additional parameter to use with it: https://registry.terraform.io/providers/vmware/vcd/latest/docs#saml_adfs
- This section of the docs should describe the new service account option too

CC: @Didainius @adambarreiro @dataclouder

Reply:
I updated the docs, mostly, to stay consistent but this part:

> saml_adfs should state which additional parameter to use with it: https://registry.terraform.io/providers/vmware/vcd/latest/docs#saml_adfs

Do you mean we should mention `saml_adfs_rpt_id` should be used with it?

Reply:
Updated, let me know if the current way is okay.

Reply:
@adambarreiro @Didainius would you review this change too:
https://github.com/vmware/terraform-provider-vcd/pull/1040/files#diff-4839085b71de83d7294472f5f8fa6652faf4b0b24df211c8eaf945ac96933aa5R305-R311

Reply:
LGTM!

Reply:
LGTM